discovery/db_discovery.go

```diff
@@ -221,3 +227,63 @@ func pmTicketParams(params *net.TicketParams) *pm.TicketParams {
 		Seed:              new(big.Int).SetBytes(params.Seed),
 	}
 }
+
+func (dbo *DBOrchestratorPoolCache) PollOrchestratorInfo(bcast common.Broadcaster) {
+	ticker := getTicker()
+	go func() {
+		for range ticker.C {
+			dbo.pollOrchestratorInfo(bcast)
+		}
+	}()
+}
+
+func (dbo *DBOrchestratorPoolCache) pollOrchestratorInfo(bcast common.Broadcaster) {
```

Comment:
Looks like most of the logic in this method is duplicated from `cacheDBOrchs()`. I think we can reuse some of the existing code - something along the lines of:

- Updating `cacheDBOrchs()` to accept `orchs []*common.DBOrch` instead of `orchs []*lpTypes.Transcoder`. We would also remove the loop that converts each `*lpTypes.Transcoder` into a `*common.DBOrch`
- Updating `CacheTranscoderPool()` to convert the list of orchestrators fetched from the contract into `[]*common.DBOrch` which can be passed to `cacheDBOrchs()`
- Updating this method to pass `orchs` into `cacheDBOrchs()`

What do you think?

Reply:
Is 7f8d8795 sorta what you're thinking of here?

Reply:
Yep!